src/Query/Filter/TextField.ts

```diff
@@ -8,23 +8,40 @@ import { FilterOrErrorMessage } from './Filter';
  * value, such as the description or file path.
  */
 export abstract class TextField extends Field {
+    private maybeNegate(match:boolean, filterMethod:String) {
+        return filterMethod.match(/not/) ? !match : match;
+    }
     public createFilterOrErrorMessage(line: string): FilterOrErrorMessage {
         const result = new FilterOrErrorMessage();
         const match = Field.getMatch(this.filterRegexp(), line);
         if (match !== null) {
             const filterMethod = match[1];
-            if (filterMethod === 'includes') {
+            if (['includes', 'does not include'].includes(filterMethod)) {
                 result.filter = (task: Task) =>
-                    TextField.stringIncludesCaseInsensitive(
-                        this.value(task),
-                        match[2],
+                    this.maybeNegate(
+                        TextField.stringIncludesCaseInsensitive(
+                            this.value(task),
+                            match[2],
+                        ),
+                        filterMethod
                     );
-            } else if (match[1] === 'does not include') {
-                result.filter = (task: Task) =>
-                    !TextField.stringIncludesCaseInsensitive(
-                        this.value(task),
-                        match[2],
+            } else if (['regex matches', 'regex does not match'].includes(filterMethod)) {
+                // Trim the leading and trailing '/'
+                const regexPattern =
+                    /\/((?![*+?])(?:[^\r\n\[/\\]|\\.|\[(?:[^\r\n\]\\]|\\.)*\])+)\/((?:g(?:im?|mi?)?|i(?:gm?|mg?)?|m(?:gi?|ig?)?)?)/;
```

Comment:
@claremacrae Linter won't shut up about this. I'm not sure what to do about it. This is a solid regexp for matching JavaScript RegExp, courtesy of stackoverflow (I had the link in the comment but must've goofed in my version control...).